### DISCUSSION TITLE
how to set workspacefolder in filesystem mcpServer

### DISCUSSION THREAD

**Discussion by wss29** [eclipse-theia/theia#14679] [Q&A]
Hi,
   I have seen demo of how to use mcp server from [theia-ai-support-mcp](https://eclipsesource.com/blogs/2024/12/19/theia-ide-and-theia-ai-support-mcp/)
![image](https://github.com/user-attachments/assets/69d99fd4-4969-43dc-9637-7791aed74dcd)
I also want to set my workspace as args of filesystem mcp server
![image](https://github.com/user-attachments/assets/e69f2696-4ab3-4e40-b1dc-67a4165aa902)
but it does not work, what is the correct format?

**Comment by JonasHelming** [comment]
In the demo, I had manually added the path to my workspace to the available directories. However, we are considering adding the workspace automatically a an option. Can you create a ticket for this?

**Reply by wss29** [reply]
hi @JonasHelming 
   I create an issue [14689](https://github.com/eclipse-theia/theia/issues/14689)

**Comment by JonasHelming** [comment]
Please note that there are Theia native tool functions to access the workspace already, see the prompt of the workspace agent

**Reply by wss29** [reply]
The workspace agent is capable of performing the same tasks. If the MCP server can access the workspace folder, we can create a custom server to handle specific or specialized operations.

**Reply by JonasHelming** [reply]
I am not sure I can follow, can you provide more details on your question/comment, please?
